### PR TITLE
Add support for @tsconfig/ember 3.0.0

### PR DESCRIPTION
**Pull request by mayatron**
- Add support for @tsconfig/ember 3.0.0
- Addresses https://github.com/san650/ember-cli-page-object/issues/613

### CODE REVIEW

File: test-app/tests/unit/-private/properties/create-test.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ module('create', function () {
       // @ts-expect-error violate types to check if it fails in weakly-typed envs
       create('');
       assert.true(false, 'should error');
-    } catch (e) {
+    } catch (e: any) {
```
Comment (ro0gr) [inline test-app/tests/unit/-private/properties/create-test.ts:97]: can we have `Error` instead of `any`?

File: test-app/package.json [hunk 2]
```diff
@@ -36,6 +36,7 @@
     "@embroider/test-setup": "^1.8.0",
     "@glimmer/component": "^1.1.2",
     "@glimmer/tracking": "^1.1.2",
+    "@tsconfig/ember": "^2.0.0",
```
Comment (ro0gr) [inline test-app/package.json:39]: looks like we should remove this from the package.json or from the ember-try scenario. I tend to remove it from the `package.json` in scope of the current `test-app`

**Review by ro0gr — COMMENTED** [review]
Overall looks great!

I'd just like to keep testing against the pre-existing `moduleResolution: "node"` + add the new `moduleResolution` test which is broken now.

File: test-app/tsconfig.json [hunk 3]
```diff
@@ -1,47 +1,18 @@
 {
+  "extends": "@tsconfig/ember/tsconfig.json",
   "compilerOptions": {
-    "target": "es2020",
-    "allowJs": true,
-    "moduleResolution": "node",
```
[the remark this answers is not in the document]
Comment (mayatron) [reply test-app/tsconfig.json:5]: The `@tsconfig/ember` 2.0.0 version has `moduleResolution` set to `node` (see [here](https://www.npmjs.com/package/@tsconfig/ember/v/2.0.0)) so there's not really a change in that respect. It only changes the value to `bundler` with the 3.0.0 version ([here](https://www.npmjs.com/package/@tsconfig/ember/v/3.0.0)).

Thoughts?

File: test-app/package.json (hunk 2 above)
Comment (mayatron) [reply test-app/package.json:39]: See comment above.

File: test-app/tests/unit/-private/properties/create-test.ts (hunk 1 above)
Comment (mayatron) [reply test-app/tests/unit/-private/properties/create-test.ts:97]: We can't use the type `Error` here, due to:

`Catch clause variable type annotation must be 'any' or 'unknown' if specified.ts(1196)`

This is an annoying issue with TypeScript, see a [long discussion thread here](https://github.com/microsoft/TypeScript/issues/20024). Alternatively, and probably more correct, we can type this as `unknown`.

**Comment by ro0gr** [comment]
@mayatron I do like your change. I'm going to merge and release it tomorrow.

**Review by ro0gr — APPROVED** [review]

**Comment by ro0gr** [comment]
Just published it as 2.1.0. Thanks for your effort!

**Pull request merged**